src/renderers/webgl/WebGLShadowMap.js

```diff
@@ -70,10 +71,12 @@ function WebGLShadowMap( _renderer, _lights, _objects, capabilities ) {
 
 		var useMorphing = ( i & _MorphingFlag ) !== 0;
 		var useSkinning = ( i & _SkinningFlag ) !== 0;
+		var useInstancing = ( i * _InstancingFlag ) !==0;
```

Comment:
Hey, this is actually a bit more complicated now that i think of it. With the constructor, the mechanism for distribution is super simple.

What happens when you set only one transformation

    setPositionOf( 20 , ... );
    setRotationOf( 20 , ... );
    //move to next one, what should happen here
    setPositionOf( 21 , ...)

I'm not keeping any position rotation scales as objects, it just goes straight to the buffer.

It would probably have to be something like 

   instanced.setPositionOf( index , vector , writeIndexToAttribute );

So:

    instanced.setPositionOf( 20 , vec , false );
    instanced.setRotationOf( 20 , vec , true );
    instanced.setPositionOf( 21 ...);